homeassistant/components/zha/light.py

```diff
@@ -452,16 +462,21 @@ async def async_get_state(self):
                 attributes, from_cache=False
             )
 
-            color_temp = results.get("color_temperature")
-            if color_temp is not None:
-                self._color_temp = color_temp
-
-            color_x = results.get("current_x")
-            color_y = results.get("current_y")
-            if color_x is not None and color_y is not None:
-                self._hs_color = color_util.color_xy_to_hs(
-                    float(color_x / 65535), float(color_y / 65535)
-                )
+            color_mode = results.get("color_mode")
+            if color_mode is not None:
```

Comment:
It would be interesting to know how many bulbs do not support this attribute and what the behavior should be if it is missing

Reply:
I also thought about this, but I did try this on some Gledopto Pro CCT only controllers which still return 0x2.
Since the native Hue integration (and maybe even Z2M and deCONZ) also seem to "require" (?) this, I'm under the impression that *all* lights (with CCT and/or RGB) support this.

The behaviour if it's missing would only mean that current_x/current_y and color_temperature aren't polled (and left at the previous state).
They'll still work fine when set with the UI.

If color_mode is None, I'm not sure if we should even try to change this mentioned behaviour.
Perhaps we could check if only current_x/current_y OR color_temperature are present and then set the state accordingly?

Reply:
Well, even Ledvance CCT light has this attribute, so I think it should be fine.

Reply:
Yeah, even my old (ZLL) OSRAM Lightify Gardenpoles (which constantly drop off the network) also support this.